libs/libc/net/lib_getifaddrs.c

```diff
@@ -0,0 +1,280 @@
+  if (sockfd < 0)
+    {
+      return sockfd;
+    }
+
+  for (i = 1, *addrs = NULL; i <= MAX_IFINDEX; i++)
+    {
+      unsigned int flags;
+      struct lifreq req =
+      {
+      };
+
+      req.lifr_ifindex = i;
+      if (ioctl(sockfd, SIOCGIFNAME, (unsigned long)&req) < 0)
+        {
+          continue; /* Empty slot, try next one */
+        }
+
+      if (ioctl(sockfd, SIOCGIFFLAGS, (unsigned long)&req) < 0)
+        {
+          goto err;
+        }
+
+      flags = req.lifr_flags;
+
+      if (myaddrs != NULL)
+        {
+          myaddrs->addrs.ifa_next = lib_zalloc(sizeof(*myaddrs));
+          myaddrs = (FAR struct myifaddrs *)myaddrs->addrs.ifa_next;
+        }
+      else
+        {
+          *addrs = lib_zalloc(sizeof(*myaddrs));
+          myaddrs = (FAR struct myifaddrs *)*addrs;
+        }
+
+      if (myaddrs == NULL)
+        {
+          goto err;
+        }
+
+      myaddrs->addrs.ifa_name = myaddrs->name;
+      strncpy(myaddrs->name, req.lifr_name, IF_NAMESIZE);
+
+      myaddrs->addrs.ifa_flags = flags;
+
+#ifdef CONFIG_NET_IPv4
+      if (ioctl(sockfd, SIOCGIFADDR, (unsigned long)&req) >= 0)
+        {
+          myaddrs->addrs.ifa_addr = (FAR struct sockaddr *)&myaddrs->addr;
+          memcpy(&myaddrs->addr, &req.lifr_addr, sizeof(req.lifr_addr));
+
+          if (ioctl(sockfd, SIOCGIFNETMASK, (unsigned long)&req) >= 0)
+            {
+              myaddrs->addrs.ifa_netmask =
+                     (FAR struct sockaddr *)&myaddrs->netmask;
+              memcpy(&myaddrs->netmask,
+                     &req.lifr_netmask, sizeof(req.lifr_netmask));
+            }
+
+          if (ioctl(sockfd, SIOCGIFDSTADDR, (unsigned long)&req) >= 0)
+            {
+              myaddrs->addrs.ifa_dstaddr =
+                     (FAR struct sockaddr *)&myaddrs->dstaddr;
+              memcpy(&myaddrs->dstaddr,
+                     &req.lifr_dstaddr, sizeof(req.lifr_dstaddr));
+            }
+          else if (ioctl(sockfd, SIOCGIFBRDADDR, (unsigned long)&req) >= 0)
+            {
+              myaddrs->addrs.ifa_broadaddr =
+                     (FAR struct sockaddr *)&myaddrs->broadaddr;
+              memcpy(&myaddrs->broadaddr,
+                     &req.lifr_broadaddr, sizeof(req.lifr_broadaddr));
+            }
+
+          if (ioctl(sockfd, SIOCGIFHWADDR, (unsigned long)&req) >= 0)
+            {
+              myaddrs->addrs.ifa_data = &myaddrs->hwaddr;
+              memcpy(&myaddrs->hwaddr,
+                     &req.lifr_hwaddr, sizeof(req.lifr_hwaddr));
+            }
+
+          myaddrs = NULL; /* Zero out to handle the dual stack */
+        }
+#endif
+
+#ifdef CONFIG_NET_IPv6
+      if (ioctl(sockfd, SIOCGLIFADDR, (unsigned long)&req) >= 0)
+        {
+          if (myaddrs == NULL)
+            {
+              myaddrs->addrs.ifa_next = lib_zalloc(sizeof(*myaddrs));
+              myaddrs = (FAR struct myifaddrs *)myaddrs->addrs.ifa_next;
```

Comment:
This does not make sense to me: `if (myaddrs == NULL)` -> `myaddrs->addrs.ifa_next`. I mean dereferencing a NULL pointer.

Reply:
You are right I need a flags to indicate this case.

Reply:
Done.